tests/unit/test_bruteforce.cpp

```diff
@@ -344,7 +344,7 @@ TYPED_TEST(BruteForceTest, brute_force_reindexing_same_vector) {
     ASSERT_EQ(VecSimIndex_IndexSize(index), 0);
 
     // The vector block should be removed.
-    ASSERT_EQ(bf_index->getVectorBlocks().size(), 0);
+    ASSERT_EQ(bf_index->getVectorsContainer()->size(), 0);
```

Comment:
can't comment in the relevant location.
I have verfied that all the friend declaration for the tests are still needed and `brute_force_reindexing_same_vector_different_id` is redundant, probably not related to this PR but can be fixed on the way :)